### PR TITLE
ci: disable telemetry for ci tests

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-374]

## Description

Reduce telemetry usage for ci tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RM-374]: https://hpe-aiatscale.atlassian.net/browse/RM-374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/devcluster/oauth.devcluster.yaml [hunk 1]
```diff
@@ -29,6 +29,8 @@ stages:
           initial_user_password: $INITIAL_USER_PASSWORD
           authz:
             rbac_ui_enabled: true
+        telemetry:
```
Comment (hamidzr) [inline .circleci/devcluster/oauth.devcluster.yaml:32]: did they have the api key needed to report? https://github.com/determined-ai/determined/blob/3baece2ff30bc4cf3ec1f54bfce63902791882b0/master/pkg/config/telemetry.go#L6

Comment (kkunapuli) [reply .circleci/devcluster/oauth.devcluster.yaml:32]: I have no idea. What's the impact either way?

Comment (carolinaecalderon) [reply .circleci/devcluster/oauth.devcluster.yaml:32]: if it defaults to false, perhaps change the name of the PR to better reflect that
don't see an issue to merging anyway though

File: e2e_tests/tests/deploy/etc/master.yaml [hunk 2]
```diff
@@ -14,3 +14,6 @@ db:
   # password: database_password
 
 launch_error: false
+
+telemetry:
+  enabled: false
```
Comment (carolinaecalderon) [inline e2e_tests/tests/deploy/etc/master.yaml:19]: missing blank line at end?

**Review by carolinaecalderon — APPROVED** [review]
LGTM

File: .circleci/devcluster/oauth.devcluster.yaml (hunk 1 above)
Comment (kkunapuli) [reply .circleci/devcluster/oauth.devcluster.yaml:32]: It defaults to true. https://docs.determined.ai/latest/reference/deploy/master-config-reference.html#id29 

Infra noticed pretty quick growth in segment data coming from ci: https://hpe-aiatscale.slack.com/archives/C03JGLKTBEJ/p1720548565903099 

Good point on the API key; I don't know how it was successfully reported.

**Review by hamidzr — APPROVED** [review]

**Pull request merged**